dot/sync/fullsync_handle_block.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ChainSafe/gossamer/dot/telemetry"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/database"
+	"github.com/ChainSafe/gossamer/lib/common"
+	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var blockSizeGauge = promauto.NewGauge(prometheus.GaugeOpts{
+	Namespace: "gossamer_sync",
+	Name:      "block_size",
+	Help:      "represent the size of blocks synced",
+})
+
+type (
+	// Telemetry is the telemetry client to send telemetry messages.
+	Telemetry interface {
+		SendMessage(msg json.Marshaler)
+	}
+
+	// StorageState is the interface for the storage state
+	StorageState interface {
+		TrieState(root *common.Hash) (*rtstorage.TrieState, error)
+		sync.Locker
+	}
+
+	// TransactionState is the interface for transaction queue methods
+	TransactionState interface {
+		RemoveExtrinsic(ext types.Extrinsic)
+	}
+
+	// BabeVerifier deals with BABE block verification
+	BabeVerifier interface {
+		VerifyBlock(header *types.Header) error
+	}
+
+	// FinalityGadget implements justification verification functionality
+	FinalityGadget interface {
+		VerifyBlockJustification(common.Hash, uint, []byte) (round uint64, setID uint64, err error)
+	}
+
+	// BlockImportHandler is the interface for the handler of newly imported blocks
+	BlockImportHandler interface {
+		HandleBlockImport(block *types.Block, state *rtstorage.TrieState, announce bool) error
+	}
```

Comment:
nit: move this to fullsync.go

Reply:
I don't think that this should be in the full sync given that this interfaces are only needed by the importer